content/guide/core/file-system.md

```diff
@@ -0,0 +1,747 @@
+
+const cache = knownFolders.temp()
+```
+
+### Accessing the root folder for the app
+
+```ts
+import { knownFolders } from '@nativescript/core'
+
+const appRootFolder = knownFolders.currentApp()
+```
+
+### Creating a folder
+
+To create a folder, call the `getFolder()` method on an instance of the `Folder`(any of the root folders above or one you create) class and  pass it the name of the folder.
+
+```ts
+folder.getFolder("folder-name")
+```
+
+You can also use the [fromPath](#frompath) static method of the Folder class.
+
+```ts
+const folderPath = path.join(knownFolders.documents().path, 'music')
+const folder: Folder = Folder.fromPath(folderPath)
+```
+
+### Renaming a folder
+
+To rename a folder, use the `rename` or `renameSync` method:
+
+```ts
+folder
+  .rename(newName)
+  .then(res => {
+    // Folder Successfully renamed.
+
+  })
+  .catch(err => {
+    //Folder couldn't be renamed
+    console.error(err)
+  })
+```
+### Check if a folder exists
+
+```ts
+const folderExists: boolean = Folder.exists('folder-path')
+```
+
+### Accessing a folder's content
+
+To get a folder's files and folders, use the `getEntitiesSync` or  `getEntities` method:
+
+
+```ts
+folder
+  .getEntities()
+  .then((entities: FileSystemEntity[]) => {
+    // do something
+  })
+  .catch(err => {
+    console.log(err)
+  })
+
+  ```
+
+### Deleting the contents of a Folder
+
+To delete all the content of a Folder, use the [clear](#clear) or [clearSync](#clearsync) method of a Folder instance:
+
+```ts
+folder.clear().then((result)=>{
+// successful delete
+}).catch((rerror)=>{
+  
+})
+
+// OR
+
+folder.clearSync((error)=>{
+
+})
+```
+
+### Deleting a folder
+
+To delete a folder, use the `remove` or `removeSync` method:
+
+```ts
+folder.remove().then((value: boolean)=>{
+
+console.log(value)
+
+}).catch((error)=>{
+
+console.error(error.message)
+})
+```
+
+### Creating, writing to and reading from a text file
+<!-- Preview: https://stackblitz.com/edit/nativescript-stackblitz-templates-5yndru?file=app/main-view-model.ts -->
+- To create a file, call the `getFile()` method on an instance of the `Folder` class and pass it the file name with the extension.
+
+```ts
+folder.getFile("my-text.txt")
+//or 
+const filePath = path.join(folder.path, 'my-text.txt')
+const file = File.fromPath(filePath)
+```
+
+- To save a text to a file, use the `writeText` or `writeTextSync` method:
+
+```ts
+file
+  .writeText('Some text')
+  .then(result => {
+    // Succeeded writing to the file.
+ 
+  })
+  .catch(error => {
+    console.log(error)
+  })
+
+```
+- To read data from a text file, use the `readText` or `readTextSync` method of the file instance:
+```ts
+   file.readText().then(res => {
+      // Succeeded read from file.
+      
+    })
+    .catch(error => {
+
+  })
+```
+### Check if a file exists
+
+To check if a file exists, call the static [exists](#exists) method of the File class and pass it the file path string:
```

Comment:
Typo.

"pass it the file path string"

Should be "pass it to the file path stored in a string"

Reply:
I think it should be "pass it the file path as a string"